go/vt/vtgate/engine/send.go

```diff
@@ -0,0 +1,94 @@
+package engine
+
+import (
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/vt/key"
+	"vitess.io/vitess/go/vt/proto/query"
+	vtrpcpb "vitess.io/vitess/go/vt/proto/vtrpc"
+	"vitess.io/vitess/go/vt/vterrors"
+	"vitess.io/vitess/go/vt/vtgate/vindexes"
+
+	querypb "vitess.io/vitess/go/vt/proto/query"
+)
+
+var _ Primitive = (*Send)(nil)
+
+// Send is an operator to send query to the specific keyspace, tabletType and destination
+type Send struct {
```

Comment:
I'd recommend a custom JSON marshaller that emits an additional fake `"Opcode": "Send"`, which will make the output plan more readable (look at engine/limit.go).